rfcs/text/0000-create-file-macho.md

```diff
@@ -0,0 +1,152 @@
+# 0000: Create the Mach-O sub-field of the File fieldset
+
+- Stage: **1 (draft)**
+- Date: **TBD**
+
+Create the Mach Object (Mach-O) sub-field, of the `file` or `process` top-level fieldsets. This document metadata can be used for malware research, as well as coding and other application development efforts.
+
+## Fields
+
+**Stage 0**
+
+This RFC is to create the Mach-O sub-field within the `file.` fieldset. This will include 35 sub-fields. `macho` itself is a nested
```

Comment:
I'd prefer to leave the top-level ECS fields all as `objects`.

What about defining `macho.architectures` as `nested` and having each architecture's fields defined underneath? Borrowing from what @andrewstucki proposed in https://github.com/elastic/beats/pull/24195.

```yaml
- name: macho
   title: Mach-O file information.
   type: group
   description: These fields contain macOS Mach Object (Mach-O) metadata.
   fields:
     - name: architectures
        description: Object files contained inside this file by architecture
        type: nested
        fields:
          - name: cpu
             description: CPU architecture target for the file.
             type: keyword
             example: 64-bit

          - name: byte_order
             description: Byte order for the file.
             type: keyword
             example: little-endian

          - name: type
             description: Mach-O file type.
             type: keyword
```

Reply:
So, you're suggesting changing:
- `macho.cpu.{architecture,byte_order,type,...}` to `macho.architecture.{cpu,byte_order,type,...}`?
- `macho.*` being a group instead of nested and `macho.architecture.cpu{cpu,byte_order,type,...}` being nested?

Reply:
Yes, mostly. One difference - I'm proposing `macho.architecture` being the nested field that all the other fields would fall under for each architecture type.

Reply:
The goal is that the "interface" to ELF, MachO, and PE have a similar feel. This will make building analytics and understanding of the data model. I discussed with @andrewstucki a while back about means to do that. The problem is that MachO fat binaries are literally two fully, self-sufficient executables in one big MachO wrapper. This means that _any_ field that we decide to nest as architecture specific will have to contain all the architecture-dependent data. The result of this discussion is that it probably made the most sense to make `file.macho` as a nested type, which drastically simplifies the data.

We should meet to discuss.